src/main.rs

```diff
@@ -289,7 +289,24 @@ fn main_try() -> Result<()> {
 
     log::info!("Protocol speed {} kHz", protocol_speed);
 
-    let mut session = probe.attach(chip).context("failed attaching to target")?;
+    let mut session = if config.general.connect_under_reset {
+        probe
+            .attach_under_reset(chip)
+            .context("failed attaching to target")?
+    } else {
+        let potential_session = probe.attach(chip);
+        match potential_session {
+            Ok(session) => session,
+            Err(session) => {
+                log::info!("The target seems to be unable to be attached to.");
+                log::info!(
+                    "A hard reset during attaching might help. This will reset the entire chip."
+                );
+                log::info!("Set `general.connect-under-reset` to enable this feature.");
```

Comment:
Maybe mention that this should be set in the configuration file? And is the option called `connect-under-reset`or `connect_under_reset`?

Reply:
Yeah, this is a misstyping :) Should be `_` ofc. I think it's pretty obvious that this should go into the configuration file just like all other options :)